src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryStrayTracker.kt

```diff
@@ -0,0 +1,315 @@
+                }
+
+                // Golden Strays, hoard/stampede
+                if (loreLine == "§7You caught a stray §6§lGolden Rabbit§7! §7A hoard of §aStray Rabbits §7has appeared!") {
+                    incrementGoldenType("stampede")
+                }
+
+                // Golden Strays, El Dorado "glimpse" - 1/3 before capture
+                goldenStrayDoradoEscape.matchMatcher(loreLine) {
+                    incrementRarity("legendary", group("amount").formatLong())
+                    incrementGoldenType("dorado")
+                }
+
+                // Golden Strays, El Dorado caught - 3/3
+                if (goldenStrayDoradoCaught.matches(loreLine)) {
+                    incrementRarity("legendary")
+                    tracker.modify { t -> t.goldenTypesCaught["dorado"] = 3 }
+                }
+
+                // Golden Strays, El Dorado (duplicate catch)
+                goldenStrayDoradoDuplicate.matchMatcher(loreLine) {
+                    incrementRarity("legendary", group("amount").formatLong())
+                    tracker.modify { t ->
+                        t.goldenTypesCaught["dorado"] = t.goldenTypesCaught["dorado"]?.plus(1) ?: 4
+                    }
+                }
+            }
+        }
+        InventoryUtils.getItemsInOpenChest().filter {
+            claimedStraysSlots.contains(it.slotIndex)
+        }.forEach {
+            if (!strayCaughtPattern.matches(it.stack.name)) {
+                claimedStraysSlots.removeAt(claimedStraysSlots.indexOf(it.slotIndex))
+            }
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onSlotClick(event: GuiContainerEvent.SlotClickEvent) {
+        if (!isEnabled()) return
+        if (event.slot == null || event.slot.slotIndex == -999) return
+        if (!InventoryUtils.getItemsInOpenChest().any { it.slotNumber == event.slot.slotNumber }) return
+        if (claimedStraysSlots.contains(event.slot.slotIndex)) return
+        try {
+            val clickedSlot = InventoryUtils.getItemsInOpenChest().first {
+                it.hasStack && it.slotNumber == event.slot.slotNumber
+            }
+            val clickedStack = clickedSlot.stack
+            val nameText = (if (clickedStack.hasDisplayName()) clickedStack.displayName else clickedStack.itemName)
+            if (nameText.equals("§6§lGolden Rabbit §8- §aSide Dish")) {
+                claimedStraysSlots.add(event.slot.slotIndex)
+                incrementGoldenType("sidedish")
+                incrementRarity("legendary", 0)
+                DelayedRun.runDelayed(1.seconds) {
+                    claimedStraysSlots.remove(claimedStraysSlots.indexOf(event.slot.slotIndex))
+                }
+            }
+        } catch (e: NoSuchElementException) {
+            return
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        tracker.firstUpdate()
```

Comment:
firstUpdate should only get called once. see the other usages of this function

Reply:
Maybe on the cf inventory open event?

Reply:
Fixed in latest commit.